core/src/apps/cardano/layout.py

```diff
@@ -122,8 +122,12 @@ async def show_warning_tx_different_staking_account(
     page1.normal("the current account.")
 
     page2 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
-    page2.normal("Staking account:")
-    page2.bold("#%d" % to_account_number(staking_account_path))
+    page2.normal(
+        "%s:"
+        % join_with_path_as_account_number("Staking account", staking_account_path)
+    )
+    page2.normal("Staking account%s:" % to_account_number_2(staking_account_path, " #"))
```

Comment:
Here I use both `join_with_path_as_account_number` and `to_account_number_2` to demonstrate their usage. I am not too happy with either, although I think I prefer `to_account_number2`.

Reply:
I would refactor `to_account_number2` to return directly the account number prefixed with `#` - I don't see why would we want to display account numbers somewhere with the hashtag and somewhere without it as this would potentially confuse users

Reply:
Since the `to_account_number` function now raises a `ValueError` when it fails, IMHO we don't need to deal with formatting.